### ISSUE TITLE
npm install failing on gulp-sass

### ISSUE THREAD

**Issue by jrhite** [dlmanning/gulp-sass#418]
I'm trying to npm install the latest version of gulp-sass. But it's failing on node-sass, via gulp-sass. The failure output is:
```
node-sass@3.4.2 install /myapp/node_modules/node-sass
> node scripts/install.js
```
My `package.json` file is below. Note that nowhere am I directly including node-sass, but it is included via the gulp-sass dependency. However, in the output above note the directory structure that it tries to run the install.js script from: `/myapp/node_modules/node-sass`. But that directory won't exist, because node-sass is a sub-dependency. Really npm should be running the install script from: `/myapp/node_modules/gulp-sass/node_modules/node-sass`

```
{
  "name": "myapp",
  "version": "1.0.0",
  "description": "",
  "main": "gulpfile.js",
  "dependencies": {},
  "devDependencies": {
    "del": "^2.0.2",
    "gulp": "^3.9.0",
    "gulp-angular-filesort": "^1.1.1",
    "gulp-bower": "0.0.10",
    "gulp-concat": "^2.6.0",
    "gulp-debug": "^2.1.2",
    "gulp-filter": "^3.0.1",
    "gulp-inject": "^3.0.0",
    "gulp-load-plugins": "^1.0.0",
    "gulp-sass": "^2.1.0",
    "gulp-util": "^3.0.7",
    "main-bower-files": "^2.9.0",
    "run-sequence": "^1.1.4",
    "wrench": "^1.5.8"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

It seems like gulp-sass and node-sass are not playing well together. Any thoughts on the best way to work around this?

**Comment by jrhite** [comment]
I should note, there are 3 related environments I'm working with here. In all 3 environments I run `npm install` followed by `gulp`.

1. my local laptop running Mac OS X
2. in a local Docker Machine `docker-config build` on my Mac 
3. in a remote Docker Machine on Digital Ocean

Everything works fine locally. Both the `npm install` and `gulp`.

For my local Docker Machine, `npm install` succeeds, but `gulp` fails with `Error: The `libsass` binding was not found...`. I can fix this by running `node ./node_modules/gulp-sass/node_modules/node-sass/scripts/install.js` before running gulp.

In the remote Docker Machine, `npm install` fails. To get further I need to remove the extra `node ./node_modules/gulp-sass/node_modules/node-sass/scripts/install.js` to get further. But then it fails with the error mentioned in my original post.

The fact that Docker installs are inconsistent and failing in different ways locally and remotely is completely puzzling me.

**Comment by LPGhatguy** [comment]
I had this same issue on a Windows 10 machine just today!

For me, upgrading to a newer version of `npm` fixed the issue -- I was running npm 3.3.7, distributed with Node 5.3.0 and upgraded to npm 3.5.2 via `npm install npm -g`.

**Comment by jrhite** [comment]
This turned out to be an issue on my end. I was going down the wrong path as I'd see several other posts around with a similar issue.

My issue turned out to be I wasn't 'dockerignoring' the node_modules directory. So in my local Docker Machine docker-config build on my Mac, it was failing because the host's node_modules directory was getting copied in to the container.

In the remote Docker Machine on Digital Ocean I was seeing different behavior because it was getting the code from github and there was no node_modules directory being copied in to the container.
